packages/sdk/src/referral.ts

```diff
@@ -740,49 +748,61 @@ export class ReferralProvider {
       }),
     );
 
-    const txs: VersionedTransaction[] = [];
-    const chunkedInstructions = chunk(claimInstructionParams, 5);
+    const chunkedInstructions = chunk(claimInstructionParams, 4);
 
-    chunkedInstructions.forEach(async (chunkParams) => {
-      let instructions: TransactionInstruction[] = [];
+    const txs: VersionedTransaction[] = await Promise.all(
+      chunkedInstructions.map(async (chunkParams) => {
+        let instructions: TransactionInstruction[] = [];
 
-      for (const {
-        referralTokenAccountPubKey,
-        projectAdminTokenAccount,
-        partnerTokenAccount,
-        mint,
-        preInstructions,
-        tokenProgramId,
-      } of chunkParams) {
-        const tx = await this.program.methods
-          .claim()
-          .accounts({
-            payer: payerPubKey,
-            project: referralAccount.project,
-            admin: project.admin,
-            projectAdminTokenAccount,
-            referralAccount: referralAccountPubKey,
-            referralTokenAccount: referralTokenAccountPubKey,
-            partner: referralAccount.partner,
-            partnerTokenAccount: partnerTokenAccount,
-            mint,
-            tokenProgram: tokenProgramId,
-          })
-          .preInstructions(preInstructions)
-          .transaction();
-        instructions.push(...tx.instructions);
-      }
+        await Promise.all(
+          chunkParams.map(
+            async (
+              {
+                referralTokenAccountPubKey,
+                projectAdminTokenAccount,
+                partnerTokenAccount,
+                mint,
+                preInstructions,
+                tokenProgramId,
+              },
+              index,
+            ) => {
+              const tx = await this.program.methods
+                .claim()
+                .accounts({
+                  payer: payerPubKey,
+                  project: referralAccount.project,
+                  admin: project.admin,
+                  projectAdminTokenAccount,
+                  referralAccount: referralAccountPubKey,
+                  referralTokenAccount: referralTokenAccountPubKey,
+                  partner: referralAccount.partner,
+                  partnerTokenAccount: partnerTokenAccount,
+                  mint,
+                  tokenProgram: tokenProgramId,
+                })
+                .preInstructions(preInstructions)
```

Comment:
i think you can do .instruction and get back an instruction and feed it back to the link i shared from solana.